### DISCUSSION TITLE
Expand on Click for Galleries is Clunky and Inefficient - Toggle for Entire Gallery?

### DISCUSSION THREAD

**Discussion by pixelvine** [WordPress/gutenberg#61259] [General]
I appreciate being able to enable lightbox functionality on images, however this really needs to have an option to apply Expand on Click to the whole gallery, not just on an individual image basis. 

I'm having trouble thinking of any instance where I would have a gallery on a post or page, but only want lightbox functionality for a select few of the images in that gallery. It would totally confuse and frustrate the user if only some of the images were clickable.

The current workflow is super inefficient. Lets say you have 25 images in your gallery. It takes 4 clicks per image to access and enable the Expand on Click setting since the last update. So you will have clicked 100 times just to enable lightbox functionality for your small gallery instance.

This could be soooooo much simpler if there was a toggle to enable this for the entire gallery, resulting in 4 clicks max. Every gallery plugin or page builder I’ve ever worked with has this basic functionality.

In addition, the ability to cycle through all of the images in the gallery with arrows would be great. Currently you have to close the lightbox and click on another image.

**Comment by EusebiuOprinoiu** [comment]
Also, there should be Previous and Next buttons to change between images in a gallery.
Please take a look at WP Featherlight for a simple and elegant implementation.

The Expand on Click implementation under the Link icon is good for individual images, but for the Gallery block, we need a global option to enable or disable the lightbox for all child images. This could be a simple toggle at the Gallery block level.

**Comment by punksinatra** [comment]
To this day still waiting for this option. Would be so much useful for sites that have multiple galleries with multiple images. 

Also with this option, there is 1 less plugin to install as the basic  gallery functionality is more than enough for most people's needs

**Comment by t-hamano** [comment]
> this really needs to have an option to apply Expand on Click to the whole gallery

This functionality has been implemented in https://github.com/WordPress/gutenberg/pull/64014. This is available on the upcoming WordPress 6.8:

![image](https://github.com/user-attachments/assets/52842a8a-63a4-40c9-89bd-6ad28522e040)

> Also, there should be Previous and Next buttons to change between images in a gallery.

This is attempted to implement in https://github.com/WordPress/gutenberg/pull/62906, but we need to fix some a11y issues.
